### ISSUE TITLE
Where do you guys save the stripe secret key?

### ISSUE THREAD

**Issue by brilliantjc** [tipsi/tipsi-stripe#306]
I saved the secret key just inside of the pay function. However, people keeping telling it my app could be cracked easily, then I will be in debt. 

What does everyone else do?

Thank you!

<img width="567" alt="screen shot 2018-06-05 at 12 40 15 pm" src="https://user-images.githubusercontent.com/25623417/40998803-b37f7a4c-68bd-11e8-9d46-ae18231247ff.png">

**Comment by isnifer** [comment]
@brilliantjc basically, in the environment. For now, we are storing tokens on the backend

**Comment by brilliantjc** [comment]
Do you need to encrypt it? in what way?

**Comment by cybergrind** [comment]
No, we don't encrypt secret keys on backend side. But there is no way to get this keys via API.

**Comment by brilliantjc** [comment]
Because in China's most popular shopping website, people can easily purchase 2.1 billion app credit/coins by paying 5 RMB($0.8). 

So, just wondering do we need to concern about that?

<img width="955" alt="screen shot 2018-06-05 at 2 40 11 pm" src="https://user-images.githubusercontent.com/25623417/41004448-c040368e-68ce-11e8-98d0-c875fcb2ad63.png">

**Comment by cybergrind** [comment]
Your client app should never have the secret key and should not be able to get it.

All operations should be done on the backend side. Till you have no critical vulnerabilites that are allowing someone get inside your backend server - you save. Cannot comment about firebase as database for secure information, you should ensure that only your backend server can get this information.

**Comment by jchirik** [comment]
If you're using Firebase, use Firebase cloud functions (on the backend) with your secret key to process charges. Create payment source tokens client-side (in your app) with your publishable key.

An example of cloud functions:
https://github.com/firebase/functions-samples/blob/master/stripe/functions/index.js

**Comment by cocodk3432** [comment]
thanks i got idea how to crack it lol
